### PR TITLE
Changes and bug fixes to support shared clusters in DBR 14.2

### PR DESCRIPTION
**Pull request by ronanstokes-db**
## Proposed changes

When using a shared cluster with UC shared mode in DBR 14.2, referencing the sparkContext produces an attribute error. 
To fix this, if an error is thrown when accessing the sparkContext, we use a default parallelism of 200 unless an alternative
value was explicitly specified. 

This PR also uses the Spark SQL function element_at rather than direct array indexing due to incompatibilities with some spark versions. 

## Types of changes

What types of changes does your code introduce to dbldatagen?
_Put an `x` in the boxes that apply_

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Change to tutorials, tests or examples
- [ ] Non code change (readme, images or other non-code assets)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. 
If you're unsure about any of them, don't hesitate to ask. We're here to help! 
This is simply a reminder of what we are going to look for before merging your code._

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Submission does not reduce code coverage numbers
- [x] Submission does not increase alerts or messages from prospector / lint

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you 
did and what alternatives you considered, etc...

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,10 +3,10 @@
 ## Change History
 All notable changes to the Databricks Labs Data Generator will be documented in this file.
 
-
-
 #### Changed
 * Updated readme to include details on which versions of Databricks runtime support Unity Catalog `shared` access mode.
+* Updated code to use default parallelism of 200 when using a shared Spark session
+* Updated code to use Sparks SQL function `element_at` instead of array indexing due to incompatibility
```
Comment (howardwu-db) [inline CHANGELOG.md:9]: "Spark's"

File: tests/test_shared_env.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+@pytest.fixture(scope="class")
+def setupLogging():
+    FORMAT = '%(asctime)-15s %(message)s'
+    logging.basicConfig(format=FORMAT)
+
+
+class TestSharedEnv:
+    """Tests to simulate testing under a Unity Catalog shared environment. In a Unity Catalog shared environment with
+    the 14.x versions of the Databricks runtime, the sparkSession object does not support use of the sparkContext
+    attribute to get the default parallelism. In this cases, we want to catch errors and return a default of
```
Comment (howardwu-db) [inline tests/test_shared_env.py:17]: In this case not cases

**Review by howardwu-db — COMMENTED** [review]
Make some grammar fixes here.

I'm a bit curious what db's stance is on porting code to shared clusters. It seems like they should try to have backward compatibility built in for the missing spark context?

**Review by howardwu-db — CHANGES_REQUESTED** [review]
See requested changes as noted

**Review by howardwu-db — COMMENTED** [review]
look good

**Review by howardwu-db — APPROVED** [review]
looks good

**Comment by GeekSheikh** [comment]
It's working for me. We're testing it a little deeper but looks good. Please don't forget to bump the version.

**Review by GeekSheikh — APPROVED** [review]
please don't forget to bump the version but looks to be working on 14.3 LTS.

**Pull request merged**